draft-ietf-anima-brski-prm.md

```diff
@@ -520,127 +540,148 @@ In {{!RFC8995}}, pledges instead need to continuously request enrollment from a
+If the registrar EE certificate is provided to the Registrar-Agent(s) via configuration, the credential used for signing artifacts for a Registrar-Agent MUST correspond to the registrar EE certificate configured at that Registrar-Agent.
+This ensures that Registrar-Agent and pledge can verify the certficate and signature of the intended domain registrar. TODO("intended" domain registrar is not clear anymore when the RegAgt can change registrar between collecting and providing artifacts)
+Overall, this has operational implications when the registrar is part of a scalable framework as described in {{Section 1.3.1 of ?I-D.richardson-anima-registrar-considerations}}.
 
-According to {{Section 5.3 of !RFC8995}}, the domain registrar performs the pledge authorization for bootstrapping within his domain based on the Pledge Voucher-Request.
+According to {{Section 5.3 of !RFC8995}}, the domain registrar performs the pledge authorization for bootstrapping within its domain based on the Pledge Voucher-Request.
+For this, it MUST possess the IDevID trust anchor(s) (i.e., root or issuing CA certificate(s)) of the pledge vendor(s)/manufacturer(s).
 This behavior is retained in BRSKI-PRM.
 
-The domain registrar MUST possess and trust the IDevID (root or issuing) CA certificate 
-of the pledge vendor/manufacturer.
+In its role as EST server {{!RFC7030}}, the domain registrar MUST also possess the domain CA certificates as defined in {{Section 5.9 of !RFC8995}}.
+
+Finally, the domain registrar MUST possess the Registrar-Agent EE certificate(s) to validate agent-signed data and to provide it to the MASA.
+The registrar MAY use the certificate verified during client authentication within the TLS sessions with the Registrar-Agent;
+in this case, the registrar MUST possess the domain trust anchor (i.e., domain CA certificate) for the Registrar-Agent EE certificate to verify the certificate chain.
+Alternatively, the Registrar-Agent EE certificate(s) MAY be provided via configuration or a repository.
 
-Further, the domain registrar MUST have its own EE credentials.
 
 ### Domain Registrar with Combined Functionality
 
-A registrar with combined BRSKI and BRSKI-PRM functionality MAY detect if the bootstrapping is performed by the pledge directly (BRSKI case) or by a Registrar-Agent (BRSKI-PRM case) based on the utilized credential for client authentication during the TLS session establishment and switch switch the operational mode from BRSKI to BRSKI-PRM.
+A registrar with combined BRSKI and BRSKI-PRM functionality MAY detect if the bootstrapping is performed by the pledge directly (BRSKI case) or by a Registrar-Agent (BRSKI-PRM case) based on the utilized credentials for client authentication during the TLS session establishment and switch the operational mode from BRSKI to BRSKI-PRM.
+
+This may be supported by a specific naming in the SAN (subject alternative name) component of the Registrar-Agent EE certificate. TODO(totally unclear purpose)
```

Comment:
I had skipped this section before. I do not understand how the SAN is supposed to "support to distinguish". Line 602 clearly states how to distinguish.

Reply:
In BRSKI the pledge uses an IDevID certificate in the TLS handshake to the Registrar. In BRSKI-PRM the Registrar-Agent uses a domain issued certificate for TLS client authentication in the handshake. This already indicates that the domain registrar is contacted by a domain component (through the domain issued EE certificate). Support via the SAN was intended to make the registrar detection during the first contact in TLS explicitly known. If the SAN is something like "registrar-agent-xyz", the domain registrar already knows from the client handshake explicitly that this peer will connect with BRSKI-PRM.

Reply:
New text on SAN integrated. Unclear removed.